emcc.py

```diff
@@ -2902,9 +2902,11 @@ def run_closure_compiler():
                                use_closure_compiler=options.use_closure_compiler,
                                debug_info=debug_info,
                                symbols_file=symbols_file)
+
+    shared.configuration.get_temp_files().note(wasm2js)
+
```

Comment:
This seems wrong, isn't this file used down on line 2912?   (maybe an accident?  or unrelated change?)

Reply:
This line only causes the temp file to be marked for deletion when program exits. It is fine to use it still (we do have that pattern in other places as well)

Although I am not entirely sure why `wasm2js.js` and `wasm2js.modular.js` all of a sudden are not getting deleted, but looking at these code sites, they are leaving behind the temp files they generate.

Reply:
But we run the test suite in such way that if any files are left around the test fail...  

At the very least this is unrelated to fix in question isn't it?

Reply:
Yeah, that is exactly what is happening. After the refactor of Closure, the test suite began failing on these temp files getting leaked, and I needed to add these in. I don't quite understand why, but these `.note()`s are certainly the proper thing to do as they are missing from the calls to wasm2js and the modular version of it.

Reply:
I see.  That that is very odd.   

Isn't there a case where the wasm2js output here actually *is* the final output?  The line `final_js = wasm2js` below suggests it could be.  

At least we should understand why refactoring the closure stuff caused this issue.   was the the case the closure cleaned this up before?   Or that closure was operating in-place before and now its not?

Reply:
>  Or that closure was operating in-place before and now its not?

I believe that is the case here.

> Isn't there a case where the wasm2js output here actually _is_ the final output? The line `final_js = wasm2js` below suggests it could be.

I'm letting the CI tell me if it's so, I believe it should bark on something in a couple of minutes.